openmrs-keycloak-smart-auth/src/main/java/org/openmrs/contrib/keycloak/smart/auth/provider/SmartLaunchAuthenticator.java

```diff
@@ -160,6 +163,7 @@ private void selectPatient(AuthenticationFlowContext context) {
 				.actionTokenBuilder(context.getUriInfo().getBaseUri(), token, clientId, authSession.getTabId())
 				.queryParam(Constants.EXECUTION, context.getExecution().getId())
 				.queryParam(QUERY_PARAM_APP_TOKEN, "{tokenParameterName}")
+				.queryParam("launchType", launchType)
```

Comment:
This still needs to be moved into the token, e.g. like we do for `user` on 148. That's the only way it's guaranteed to be signed. We should also likely just pass the scopes, maybe without launch, e.g.,

```java
externalToken.setNote("launchTypes", getLaunchScopes());

//...

private String getLaunchScopes(String scope) {
	Arrays.stream("launch/patient launch/encounter".split("\\s")).filter(it -> it.startsWith("launch/")).map(it -> it.substring(6)).collect(
			Collectors.joining(" "));
}
```

Then we do need to handle the difference in OpenMRS, but at least this way we're less directly tied to knowing that, e.g., we only care about one or the other scope.

Reply:
Again, the idea with putting things in the token is that the token is signed and therefore we can verify that it was generated by Keycloak whereas a query parameter could be populated by a malicious user (this is a security context, so we have to be careful about security).

Reply:
@ibacher thanks, I will change this and put the `launchType` parameter into token.